http-clients/src/main/java/feign/OptionalAwareDecoder.java

```diff
@@ -35,11 +36,19 @@ public OptionalAwareDecoder(Decoder delegate) {
 
     @Override
     public Object decode(Response response, Type type) throws IOException, FeignException {
-        if (response.status() == 204 && Types.getRawType(type).equals(Optional.class)) {
-            return Optional.absent();
+        if (Types.getRawType(type).equals(Optional.class)) {
+            if (response.status() == 204) {
+                return Optional.absent();
+            } else {
+                return Optional.of(delegate.decode(response, getInnerType(type)));
```

Comment:
It looks like any resource that returns null from jersey will be a 204, so there's no way to write this test without mocking -- which doesn't seem useful to me.